### PR TITLE
[OCP3] allow isolated flatpak

### PR DESCRIPTION
**Pull request by rcerven**
* CLOUDBLD-7764

Signed-off-by: Robert Cerven <rcerven@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Pull request has a link to an osbs-docs PR for user documentation updates

### CODE REVIEW

File: tests/test_builder_containerbuild.py [hunk 1]
```diff
@@ -444,7 +444,7 @@ def _mock_osbs(self, koji_build_id, src, koji_task_id, orchestrator=False, sourc
         (osbs
             .should_receive('wait_for_build_to_get_scheduled')
             .with_args('os-build-id'))
-        (osbs.should_receive('cancel_build').never)  # pylint: disable=expression-not-assigned
+        (osbs.should_receive('cancel_build').never())  # pylint: disable=expression-not-assigned
```
Comment (MartinBasti) [inline tests/test_builder_containerbuild.py:447]: Please remove pylint comment, it should be fixed by your change

File: koji_containerbuild/plugins/builder_containerbuild.py [hunk 2]
```diff
@@ -961,24 +961,23 @@ def handler(self, src, target, opts=None):
...
 
-        # scratch builds do not get imported, and consequently not tagged
-        if not self.opts.get('scratch') and not flatpak:
-            self.check_whitelist(component, target_info)
-
-        if flatpak:
-            expected_nvr = None
+            # scratch builds do not get imported, and consequently not tagged
+            if not self.opts.get('scratch'):
+                self.check_whitelist(component, target_info)
```
Comment (MartinBasti) [inline koji_containerbuild/plugins/builder_containerbuild.py:980]: Is this indented correctly? Shouldn't it be one level up?

Comment (rcerven) [reply koji_containerbuild/plugins/builder_containerbuild.py:980]: we don't have dockerfile => component for flatpak, so check_whitelist is required only for non-flatpak and non-scratch

File: tests/test_builder_containerbuild.py (hunk 1 above)
Comment (rcerven) [reply tests/test_builder_containerbuild.py]: done

**Review by mkosiarc — APPROVED** [review]

**Review by MartinBasti — APPROVED** [review]

**Comment by rcerven** [comment]
rn updated

**Pull request merged**